repondeur/tests/views/test_auth.py

```diff
@@ -1,204 +1,238 @@
-
-    assert resp.status_code == 302
-    assert (
-        resp.location
-        == "https://zam.test/bienvenue?source=https%3A%2F%2Fzam.test%2Flectures%2F"
-    )
-
-    assert "auth_tkt" in app.cookies  # and now we have the auth cookie
-
-    domains = {cookie.domain for cookie in app.cookiejar}
-    assert domains == {".zam.test", "zam.test"}
-
-    for cookie in app.cookiejar:
-        assert cookie.name == "auth_tkt"
-        assert cookie.path == "/"
-        assert cookie.secure is True
-
-        # Auth cookie should expire after 7 days
-        assert cookie.expires == int(time.time()) + (7 * 24 * 3600)
-
-        # We want users to be able to follow an e-mailed link to the app
-        # (see: https://www.owasp.org/index.php/SameSite)
-        assert cookie.get_nonstandard_attr("SameSite") == "Lax"
-
-
-def test_user_gets_an_email_after_identifying_herself(app):
-    mailer = get_mailer(app.app.registry)
-
-    mailer.outbox = []  # FIXME: do that within the fixture!
-    assert len(mailer.outbox) == 0
-
-    resp = app.post("/identification", {"email": "jane.doe@exemple.gouv.fr"})
-
-    assert resp.status_code == 302
-    assert (
-        resp.location
-        == "https://zam.test/bienvenue?source=https%3A%2F%2Fzam.test%2Flectures%2F"
+    @pytest.mark.parametrize(
+        "valid_email", ["foo@exemple.gouv.fr", "BAR@EXEMPLE.GOUV.FR"]
     )
-    assert len(mailer.outbox) == 1
-    assert mailer.outbox[0].subject == "Se connecter à Zam"
-    assert (
-        mailer.outbox[0].body
-        == """\
-Bonjour,
-
-Pour vous connecter à Zam, veuillez cliquer sur l’adresse suivante :
-https://zam.beta.gouv.fr/login?token=FOO
-
-Bonne journée !"""
+    def test_user_gets_an_auth_cookie_after_identifying_herself(self, app, valid_email):
+        assert "auth_tkt" not in app.cookies  # no auth cookie yet
+
+        resp = app.post("/identification", {"email": valid_email})
+
+        assert resp.status_code == 302
+        assert (
+            resp.location
+            == "https://zam.test/bienvenue?source=https%3A%2F%2Fzam.test%2Flectures%2F"
+        )
+
+        assert "auth_tkt" in app.cookies  # and now we have the auth cookie
+
+        domains = {cookie.domain for cookie in app.cookiejar}
+        assert domains == {".zam.test", "zam.test"}
+
+        for cookie in app.cookiejar:
+            assert cookie.name == "auth_tkt"
+            assert cookie.path == "/"
+            assert cookie.secure is True
+
+            # Auth cookie should expire after 7 days
+            assert cookie.expires == int(time.time()) + (7 * 24 * 3600)
+
+            # We want users to be able to follow an e-mailed link to the app
+            # (see: https://www.owasp.org/index.php/SameSite)
+            assert cookie.get_nonstandard_attr("SameSite") == "Lax"
+
+    def test_user_gets_an_email_with_url_token_after_identifying_herself(self, app):
+
+        mailer = get_mailer(app.app.registry)
```

Comment:
What about turning this into a fixture for easy access?

Reply:
👍 good idea!